drivers/accel/adxl355/adxl355.c

```diff
@@ -0,0 +1,925 @@
+ * @brief Writes the activity count register value.
+ *
+ * @param dev       - The device structure.
+ * @param act_count - Register value.
+ *
+ * @return ret      - Result of the writing procedure.
+*******************************************************************************/
+int adxl355_set_activity_count_register(struct adxl355_dev *dev,
+					uint8_t act_count)
+{
+	int ret;
+	ret = adxl355_write_device_data(dev, ADXL355_ADDR(ADXL355_ACT_COUNT),
+					GET_ADXL355_TRANSF_LEN(ADXL355_ACT_COUNT), &act_count);
+	if (!ret)
+		dev->act_count = act_count;
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Configures the interrupt map for INT1 and INT2 pins.
+ *
+ * @param dev              - The device structure.
+ * @param interrupt_config - Interrupt mapping.
+ *
+ * @return ret             - Result of the configuration procedure.
+*******************************************************************************/
+int adxl355_configure_interrupt_pins (struct adxl355_dev *dev,
+				      union adxl355_interrupt_mask interrupt_config)
+{
+	uint8_t reg_val = interrupt_config.value;
+
+	return adxl355_write_device_data(dev, ADXL355_ADDR(ADXL355_INT_MAP),
+					 GET_ADXL355_TRANSF_LEN(ADXL355_INT_MAP),&reg_val);
+}
+
+/***************************************************************************//**
+ * @brief Sets the interrupt polarity.
+ *
+ * @param dev                - The device structure.
+ * @param interrupt_polarity - Interrupt polarity to be set.
+ *
+ * @return ret               - Result of the reading procedure.
+*******************************************************************************/
+int adxl355_set_interrupt_polarity(struct adxl355_dev *dev,
+				   enum adxl355_interrupt_polarity interrupt_polarity)
+{
+	int ret;
+	uint8_t reg_value;
+
+	ret = adxl355_read_device_data(dev, ADXL355_ADDR(ADXL355_RANGE),
+				       GET_ADXL355_TRANSF_LEN(ADXL355_RANGE), &reg_value);
+	if (ret)
+		return ret;
+
+	reg_value |= ((uint8_t)(interrupt_polarity) << 6) & BIT(6);
+	ret = adxl355_write_device_data(dev, ADXL355_ADDR(ADXL355_RANGE),
+					GET_ADXL355_TRANSF_LEN(ADXL355_RANGE), &reg_value);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Converts array of raw acceleration to uint32 data raw acceleration
+ *
+ * @param dev       - The device structure.
+*  @param raw_array - Raw acceleration array.
+ *
+ * @return ret    - Converted data.
+*******************************************************************************/
+static uint32_t adxl355_accel_array_conv(struct adxl355_dev *dev,
+		uint8_t *raw_array)
+{
+	uint32_t raw_accel;
+
+	// raw_array[0] bits [7-0]: DATA bits [19:12]
+	// raw_array[1] bits [7-0]: DATA bits [11: 4]
+	// raw_array[2] bits [7-4]: DATA bits [ 3: 0]
+	// raw_array[2] bits i[3-0]: reserved
+	raw_accel = (raw_array[0] << 16) | (raw_array[1] << 8) | raw_array[2];
```

Comment:
After https://github.com/analogdevicesinc/no-OS/pull/1260 is merged into the master branch you can use here `get_unaligned_be24`